rpcserver.go

```diff
@@ -769,32 +809,55 @@ func (r *rpcServer) Start() error {
 	//
 	// TODO(roasbeef): eventually also allow the sub-servers to themselves
 	// have a REST proxy.
-	mux := proxy.NewServeMux(customMarshalerOption)
+	restMux := proxy.NewServeMux(customMarshalerOption)
+	restCtx, restCancel := context.WithCancel(context.Background())
 
 	err := lnrpc.RegisterLightningHandlerFromEndpoint(
-		context.Background(), mux, r.restProxyDest,
-		r.restDialOpts,
+		restCtx, restMux, r.restProxyDest, r.restDialOpts,
 	)
 	if err != nil {
 		return err
 	}
+
+	// Before listening on any of the inter
+
 	for _, restEndpoint := range cfg.RESTListeners {
 		lis, err := lncfg.TLSListenOnAddress(restEndpoint, r.tlsCfg)
 		if err != nil {
-			ltndLog.Errorf(
-				"gRPC proxy unable to listen on %s",
-				restEndpoint,
-			)
+			ltndLog.Errorf("gRPC proxy unable to listen on %s",
+				restEndpoint)
 			return err
 		}
 
 		r.listenerCleanUp = append(r.listenerCleanUp, func() {
+			restCancel()
 			lis.Close()
 		})
 
+		// Before actually listening on the REST listener, give external
+		// subservers the chance to register to our REST mux. Those
+		// external subservers (think GrUB) are responsible for
+		// starting/stopping on their own, we just let them register
+		// their services to the same server instance so all of them can
+		// be exposed on the same port/listener.
+		if len(r.listeners) == 1 &&
```

Comment:
So we can only have a single external listener for the REST sub-servers? I guess this is the case already though.

Reply:
This was just me trying out a few things. We only create one instance of the `proxy.ServeMux` so we also only need one callback supplied. I refactored the subserver config to remove this restriction of only one listener.